homeassistant/helpers/service.py

```diff
@@ -109,3 +110,41 @@ def extract_entity_ids(hass, service_call, expand_group=True):
             return [service_ent_id]
 
         return service_ent_id
+
+
+@callback
+def async_get_state_services(hass, domain, state):
+    """Return a dict of services for domain that can be used to set state.
```

Comment:
Instead of a dict, we should return just the 1 service that can be used to achieve it. We should not ever have two services that can achieve the same state.